thinit.cxx

```diff
@@ -332,6 +336,11 @@ void thinit::load()
   if (loaded_ok) {
     thsplit_args(&mbf,this->path_cavern.get_buffer());
     this->path_cavern = *(mbf.get_buffer());
+    // VG 120416: Replace "aven.exe" with "cavern.exe". New Survex versions write up aven, but therion needs cavern
+    // See http://mailman.speleo.sk/pipermail/therion/2015-September/006072.html
+    int pathlen = strlen(this->path_cavern.get_buffer());
+    this->path_cavern.get_buffer()[pathlen - strlen("aven.exe")] = 0;
+    this->path_cavern.strcat("cavern.exe");
```

Comment:
This ought to check that the string actually ends with `"aven.exe"` rather than just replacing the last 8 bytes regardless of what they are - otherwise it's going to break with an older install of Survex or if the user has set a custom action to run cavern instead of aven.  In those case the patched code would change the command to run `"cacavern.exe"`.